rerun_py/src/python_session.rs

```diff
@@ -226,12 +226,20 @@ impl PythonSession {
         self.sink.send(log_msg);
     }
 
+    pub fn send_arrow_msg(&mut self, arrow_msg: ArrowMsg) {
+        let recording_id = self.recording_id().unwrap_or_default();
```

Comment:
again, I think it would be better if we actually generated a random Recording ID (and maybe remove the `RecordingId::default` footgun)

Reply:
Generating a random recording-id *here* would be problematic... if for some reason the recording-id isn't set at this point it would send every message to its own random recording, which I would argue is actually worse than using ZERO.

(For what it's worth, RecordingId::default() is also what was previously used on the server-side if no recording-id was ever received).

Either way, I think our choices are to either bail or use a predetermined but *stable* recording id. Bailing seems like a bad choice, so then we need to decide which stable RecordingId to use and how stable we want it to be. I think we basically have 3 choices:

(1) Keep using default()

(2) Introduce a fully deterministic "random"... baking in a once-off value:
```
RANDOM = RecordingId(uuid::Uuid::parse_str("74d37750-6409-4eac-a1ad-c534772c50c0")`
```

(3) or falling back on something like:
```
#[inline]
pub fn static_random() -> Self {
    use once_cell::sync::Lazy;
    static STATIC_RANDOM: Lazy<RecordingId> = Lazy::new(RecordingId::random);
    *STATIC_RANDOM
}
```

My argument in favor of sticking with default ZERO is that it makes it more clear something has gone wrong and the reality is we really don't have a recording id.  The UI correctly shows UNKNOWN if that happens somehow but things otherwise work.

Reply:
The other way of basically side-stepping the issue is to modify:

https://github.com/rerun-io/rerun/blob/b1d927b049581d60dbe946b2c7c05e79907e3b32/rerun_py/src/python_session.rs#L13-L18

And simply make recording_id non-optional with one of the 3 choices from above.  That will basically become the fallback value if somehow a user manages to create a set of cirmcumstances where they are invoking log calls but we never hit:
https://github.com/rerun-io/rerun/blob/b1d927b049581d60dbe946b2c7c05e79907e3b32/rerun_py/src/python_bridge.rs#L130

Since that happens at import-time, would have to be some pretty exotic circumstances like linking against a custom rust-library that also tries to log during the init phase?

Reply:
I think `RecordingMetaData::recording_id` should be non-optional yes, initialized to `default_recording_id(py)`

Reply:
Using `default_recording_id(py)` is kind of a can of worms... it requires that we have the `Python<_>` object (which holds the GIL) at the point when we are initializing the recording_id, which I believe means always passing it through as, e.g. `python_session(py)`... but that then means we can't acquire the python_session without holding the GIL and I'm worried about doing that as part of this change without spending some time debugging a new set of dead-locks.

I'm going to default-initialize to ZERO and create an issue.

Reply:
sounds good